### ISSUE TITLE
graphics text alternatives

### ISSUE THREAD

**Issue by pkra** [AmerMathSoc/texml#28]
WCAG, cf. https://www.w3.org/WAI/WCAG21/quickref/#text-alternative requires non-text content, in particular graphics (both raster and SVG) to have adequate text alternatives.

We currently do not provide consistent or adequate text alternatives. We also lack support for surfacing text alternatives across the tool chain.

An open list of relevant details:

* [ ] TeX sources generally lack text alternatives (as they are not required from authors) 
  * [ ] external images (i.e. `\includegraphics`'ed images) do no have text alternatives
  * [ ] "internal" non-text (e.g., tikz) lacks means of providing 
* [x] SVGs generated from TeX code have the TeX code as title (e.g. tikz, algorithms)
  * we <s>do not</s> expose those in the final products  
* [x] equation rendering using MathJax generally has the best possible text alternatives (via speech-rule-engine)
* [ ] with the exception of images within equations, e.g., when a subexpressions uses tikz we have a mixed situation - see #105

**Comment by pkra** [comment]
To gather a few pertinent comments from #55 (so that we can close that issue in favor of this one)

- Beyond the official LaTeX syntax, it would be good to provide an option for `\begin{SVG}`
- it would be good to duplicate the title of the generated SVGs in the XML to make things simpler downstream
- ideas for linking to long textual alternatives:
  - maybe just xlink:href on the graphic? 
  - Is TeX \label too TeX-y - cf. https://github.com/AmerMathSoc/texml/issues/36 
  - if the href just links to a (empty) target element, we'll have trouble working out what to really use as text alternative.

From other work we've done, macro ideas:

- `\textAlternativeLabel{}` provides a local text alternative for an associated fragment, e.g., a tikz or xypic construction. Association works as with \label.
- `\wrappedTextAlternative{}{}` provides the first argument as local text alternative for the second argument. 
- `\refTextAlternative{}{}` -  provides the cross-referenced object as text alternative for the second argument.
- `\wrappedRefTextAlternative` - provides the the cross-referenced object as local text alternative for the second argument. 
- `\detailsLabel{}` - provides additional non-visual information for an associated fragment. 
-  `\wrappedDetails{}` - provides the first argument as additional non-visual information for the second argument.
